Marketplace/Card.js

```diff
@@ -0,0 +1,284 @@
+        //get data from message
+        var data = JSON.parse(message);
+        //check if message is meant for you and is being sent to the correct card
+        if ((channel == showChannel) && (MyAvatar.sessionUUID == data[0]) && (_this.entityID == data[2])) {
+            //delete an overlay if there is one
+            try {
+                Overlays.deleteOverlay(cardOverlay);
+                cardOverlay = undefined;
+            } catch (err) {
+                //e
+            }
+            var card = Entities.getEntityProperties(_this.entityID, ['position', 'rotation', 'dimensions', 'name']);
+            //get which hand the card is in
+            hand = data[1];
+            //add overlay to card so that the person holding the card can still see it
+            cardOverlay = Overlays.addOverlay("image3d", {
+                url: "https://hifi-content.s3.amazonaws.com/jedon/Game_Creater_Toolkit/DeckOfCards/DeckOfCardsTexture/" + card.name + ".jpg",
+                subImage: { x: 1024, y: 361, width: 1024, height: 1326},
+                ignoreRayIntersection: true,
+                parentID: _this.entityID,
+                rotation: card.rotation,
+                localPosition: {
+                        x: 0,
+                        y: 0,
+                        z: .01
+                    },
+                dimensions: card.dimensions,
+                color: { red: 255, green: 255, blue: 255},
+                alpha: 1,
+                solid: true,
+                isFacingAvatar: false,
+                drawInFront: false
+            });
+        }
+    };
+
+    _this.startNearGrab = function(entityID, args) {
+        //if the glove is in your hand then check to see if its touching a card
+        var props = Entities.getEntityProperties(_this.entityID);
+        var properties = JSON.parse(props.userData);
+        var held = properties.held;
+        if (held === false) {
+            hideCard(args);
+        }
+    };
+
+    _this.startDistanceGrab = function(entityID, args) {
+        //if the glove is in your hand then check to see if its touching a card
+        var props = Entities.getEntityProperties(_this.entityID);
+        var properties = JSON.parse(props.userData);
+        var held = properties.held;
+        if (held === false) {
+            hideCard(args);
+        }
+    };
+
+    _this.releaseGrab = function (entityID, args) {
+        var hand = args[0];
+        //check if the object you are holding is a card and if it isnt already in your hand
+        var props = Entities.getEntityProperties(_this.entityID);
+        var properties = JSON.parse(props.userData);
+        var checkIfCard = properties.card;
+        var held = properties.held;
+        var deckHandlerID = properties.deckHandlerID;
+        //check to see if you are holding a card that is not already being held
+        if ((checkIfCard === true) && (held === false)) {

```

Comment:
Consolidate these into a single function that you can call for each hand -- too much duplicated code